File: Data Structures & Algorithms/remove-duplicates-from-sorted-array/submission-2.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        vector<int> nums2(201,0);
        for(int num:nums){
            if (num<0){
                nums2[abs(num)+100]++;
            }
            else{
                nums2[num]++;
            }
        }
        int k=0;
        for(int i=200;i>=101;i--){
            if(nums2[i]>=1){
                nums[k]=100-i;
                k++;
            }
        }
        for(int i=0;i<=100;i++){
            if(nums2[i]>=1){
                nums[k]=i;
                k++;
            }
        }
      
        return k;
    }
};

File: Data Structures & Algorithms/subarray-sum-equals-k/submission-2.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        res = prefix = 0
        prefixSums = { 0 : 1 }

        for num in nums:
            prefix += num
            diff = prefix - k
            if diff in prefixSums:
                res+=prefixSums[diff]

            prefixSums[prefix]=prefixSums.get(prefix,0)+1

        return res